lms/djangoapps/gradebook/tests.py

```diff
@@ -158,6 +161,27 @@ def test_receiver_on_score_changed(self):
 
     @patch.dict(settings.FEATURES, {
                 'ALLOW_STUDENT_STATE_UPDATES_ON_CLOSED_COURSE': False,
+                'SIGNAL_ON_SCORE_CHANGED': True,
+                'ENABLE_NOTIFICATIONS': True
+    })
+    @patch.dict(settings.FEATURES, {})
+    def test_notifications_publishing(self):
+        initialize_notifications()
+
+        # assert user has no notifications
+        self.assertEqual(get_notifications_count_for_user(self.user.id), 0)
+
+        self._create_course()
+        module = self.get_module_for_user(self.user, self.course, self.problem)
+        grade_dict = {'value': 0.75, 'max_value': 1, 'user_id': self.user.id}
+        module.system.publish(module, 'grade', grade_dict)
+
+        # user should have had a notification published as he/her is now in the
+        # leaderboard
+        self.assertEqual(get_notifications_count_for_user(self.user.id), 1)
```

Comment:
Can we assert that the proper notification was received in addition to simply checking that a single notification is present?  Should we also assert that no other notifications are present?  (might not be necessary)